ndt7/handler/handler.go

```diff
@@ -65,13 +70,41 @@ func (h Handler) downloadOrUpload(writer http.ResponseWriter, request *http.Requ
 	// and close(2), we'll end up keeping sockets that caused an error in the
 	// code above (e.g. because the handshake was not okay) alive for the time
 	// in which the corresponding *os.File is kept in cache.
-	defer warnonerror.Close(conn, "download: ignoring conn.Close result")
+	defer warnonerror.Close(conn, "downloadOrUpload: ignoring conn.Close result")
 	logging.Logger.Debug("downloadOrUpload: opening results file")
 	resultfp, err := results.OpenFor(request, conn, h.DataDir, kind)
 	if err != nil {
 		return // error already printed
 	}
-	defer warnonerror.Close(resultfp, "download: ignoring resultfp.Close result")
+
+	// Collect test metadata.
+	clientAddr := conn.RemoteAddr().(*net.TCPAddr)
+	serverAddr := conn.LocalAddr().(*net.TCPAddr)
```

Comment:
I am slightly confused here. Will the `*net.TCPAddr` cast _also_ work for `tls.Conn` connections?